torch/csrc/distributed/c10d/ProcessGroupGloo.cpp

```diff
@@ -308,7 +309,8 @@ ProcessGroupGloo::AsyncWork::AsyncWork(
       context_(std::move(context)),
       outputTensors_(std::move(outputTensors)),
       future_(createFutureAsOutput(outputTensors_)),
-      seq_(seq) {
+      seq_(seq),
+      timeout_(timeout == kUnsetTimeout ? context_->getTimeout() : timeout) {
```

Comment:
where is the timeout from `context_->getTimeout()` coming from?

Reply:
This is coming from the overall PG timeout that's set when the PG is created. There's a global timeout on the gloo Context